File: Eduardo/python/Maximum_Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        maxim = nums[0]
        current = nums[0]

        for i in range(1, len(nums)):

            current = max(nums[i], nums[i] + current)

            maxim = max(maxim, current)
        return maxim

File: Eduardo/c++/Maximum_Subarray.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int presentSum = nums[0];
        int globalMax = nums[0];

        for (int i = 1; i < nums.size(); i++)
        {
            presentSum = max(nums[i], nums[i] + presentSum);

            globalMax = max(globalMax, presentSum);
        }

        return globalMax;
    }
};